app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/relationships/RelationshipPresenterImpl.java

```diff
@@ -169,9 +169,21 @@ public void deleteRelationship(Relationship relationship) {
     }
 
     @Override
-    public void addRelationship(String trackEntityInstance_A, String relationshipType) {
+    public void addRelationship(String trackEntityInstance_A, String relationshipTypeUid) {
+        RelationshipType relationshipType = d2.relationshipModule().relationshipTypes().withConstraints().uid(relationshipTypeUid).blockingGet();
+
+        String fromTei;
+        String toTei;
+        if(relationshipType.bidirectional() && relationshipType.toConstraint().trackedEntityType().uid().equals(teiType)){
+            fromTei = trackEntityInstance_A;
+            toTei = teiUid;
+        }else{
+            fromTei = teiUid;
+            toTei = trackEntityInstance_A;
+        }
+
```

Comment:
any change to add unit test to this method?

Reply:
Sure